test/test_scripts/test_shard_bam.py

```diff
@@ -0,0 +1,47 @@
+import pysam
+import pathlib
+
+
+def get_read_zmw_counts(file):
+    bf = pysam.Samfile(file, 'rb', check_sq=False)
+
+    zmw_counts = {}
+    for read in bf:
+        zmw = read.get_tag("zm")
+        zmw_counts[zmw] = zmw_counts.get(zmw, 0) + 1
+
+    bf.close()
+
+    return zmw_counts
+
+
+def test_shard_bam(script_runner):
+    bam = "test/test_data/tiny.bam"
+    testdir = "test/test_output"
+    prefix = f"{testdir}/shard"
+    num_shards = 2
+
+    pathlib.Path(testdir).mkdir(parents=True, exist_ok=True)
+    ret = script_runner.run("../../docker/lr-pb/shard_bam.py", "-p", prefix, "-n", str(num_shards), bam)
```

Comment:
again a bigger topic (not to be dealt with in this PR), testing while referring to a hard-coded path is not ideal.